### ISSUE TITLE
SDDM crash on linux 6.4

### ISSUE THREAD

**Issue by noraj** [sddm/sddm#1761]
## Overview

TL;DR: SDDM crashes on vanilla kernel 6.4+ but not with linux LTS 6.1+.

Since linux 6.4.0 (6.4.1 and 6.4.2 too):

1. boot to vanilla kernel
2. enter luks password
3. systemd starts
4. when reaching the graphical target, sddm-greater crashes, I no longer have any video signal, my displays are going into sleep mode

While using linux-lts 6.1.38

1. boot to lts kernel
2. enter luks password
3. systemd starts
4. sddm greater displays correctly

## Environment info

GPU

```
➜ lspci -k | grep -A 3 -E "(VGA|3D)"
0a:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] (rev c7)
Subsystem: Micro-Star International Co., Ltd. [MSI] Radeon RX 480 Gaming X 8GB
Kernel driver in use: amdgpu
Kernel modules: amdgpu
```

System

```
➜ pacman -Qi linux | grep Version          
Version         : 6.4.2.arch1-1

➜ pacman -Qi linux-lts | grep Version      
Version         : 6.1.38-1

➜ pacman -Qi qt5-declarative | grep Version
Version         : 5.15.10+kde+r26-1

➜ pacman -Qi sddm | grep Version
Version         : 0.20.0-1

➜ pacman -Qi sddm-kcm | grep Version
Version         : 5.27.6-1

➜ grep 'PRETTY_NAME' /etc/os-release 
PRETTY_NAME="Arch Linux"
```

KDE

```
Operating System: Arch Linux 
KDE Plasma Version: 5.27.6
KDE Frameworks Version: 5.108.0
Qt Version: 5.15.10
Kernel Version: 6.1.38-1-lts (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 1600X Six-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: AMD Radeon RX 480 Graphics
Product Name: AB350 Pro4
```

## Crash info

 In `journalctl` I can sometimes see "systemd-coredump[1035]: [🡕] Process 986 (sddm-greeter) of user 975 dumped core."

https://pastebin.com/raw/W4Qfyv3w

Here is the backtrace with debug symbols captured with

```
$ sudo -E coredumpctl gdb
(gdb) set logging enabled
(gdb) bt full
```

[gdb.txt](https://github.com/sddm/sddm/files/12012916/gdb.txt)

Outside of that I don't see any errors related to (error level) from the previous boot on the logs with `journalctl -b -1`.

However, I can see some warning level logs `journalctl -b -1 -p warning` (boot-1_warnings.log) for amdgpu, sddm-greeter, sddm-helper and sddm. (the 2h timing shift in the log timestamp is due to the hardware clock and the ntp sync in between during the init).

[boot-1_warnings.log](https://github.com/sddm/sddm/files/12012931/boot-1_warnings.log)

 Full last boot (journalctl -b -1) logs (boot-1_full.log).

[boot-1_full.log](https://github.com/sddm/sddm/files/12012933/boot-1_full.log)

Note: I was using a non-default SDDM theme "Sweet" but I switched back to the default KDE Breeze one and the same crash still occurs.

**Comment by Vogtinator** [comment]
Does it crash using the default builtin theme as well?

**Comment by noraj** [comment]
> Does it crash using the default builtin theme as well?

What's the default theme? On ArchLinux the default themes packaged with [sddm](https://archlinux.org/packages/extra/x86_64/sddm/) are elarun, maldives and maya.

**Comment by Vogtinator** [comment]
maui. It's selected if you just don't set any `[Theme] Current=`.

**Comment by noraj** [comment]
> maui. It's selected if you just don't set any `[Theme] Current=`.

I test with AL default maya and it's still crashing.

I also just tested the SDDM default theme (maui) and I still have no video signal.

**Comment by thx1111** [comment]
sddm 0.20.0-1
linux 6.4.3.arch1-1
lxqt-session 1.3.0-3

This may or may not be related, but - since upgrading to sddm 0.20 -

I have `sddm` on Arch starting X11 with a default user and then starting LXQt, which segfaults on libQt5Gui.so.5, with sddm subsequently falling back to the greeter:
```
...
 sddm-helper[560]: Writing cookie to "/tmp/xauth_BkEvxc"
 sddm-helper[560]: Starting X11 session: "" "/usr/share/sddm/scripts/Xsession \"startlxqt\""
 systemd[562]: Starting Update XDG user dir configuration...
 sddm[360]: Session started true
 systemd[562]: Finished Update XDG user dir configuration.
 systemd[562]: Reached target Main User Target.
 systemd[562]: Startup finished in 1.833s.
 systemd[562]: Created slice User Core Session Slice.
 systemd[562]: Starting D-Bus User Message Bus...
 systemd[562]: Started D-Bus User Message Bus.
 lxqt-session[571]: qt.qpa.xcb: could not connect to display :0
 kernel: lxqt-session[571]: segfault at 0 ip 00007f80e333ce05 sp 00007ffc773ccc50 error 4 in libQt5Gui.so.5.15.10[7f80e32f8000+471000] likely on CPU 1 (core 1, socket 0)
 kernel: Code: 48 8b 05 5e 08 53 00 48 8b 38 48 8b 07 48 8b 40 68 48 39 d0 0f 85 0b 01 00 00 48 8b 35 4c 52 53 00 48 85 f6 0f 84 bb 00 00 00 <48> 8b 06 48 8b 40 68 48 8d 15 5d a5 17 00 48 39 d0 0f 85 d4 00 00
 systemd[1]: Created slice Slice /system/systemd-coredump.
 systemd[1]: Started Process Core Dump (PID 586/UID 0).
 systemd-coredump[587]: Process 571 (lxqt-session) of user 1001 dumped core.

                        Stack trace of thread 571:
                        #0  0x00007f80e333ce05 n/a (libQt5Gui.so.5 + 0x13ce05)
                        #1  0x00007f80e3349e66 _ZN15QGuiApplication4fontEv (libQt5Gui.so.5 + 0x149e66)
                        #2  0x00007f80e33f4a54 _ZN5QFontC1Ev (libQt5Gui.so.5 + 0x1f4a54)
                        #3  0x00007f80e44f7c43 _ZN17LXQtPlatformThemeC2Ev (libqtlxqt.so + 0x10c43)
                        #4  0x00007f80e44f7d4b n/a (libqtlxqt.so + 0x10d4b)
                        #5  0x00007f80e33342e7 _ZN21QPlatformThemeFactory6createERK7QStringS2_ (libQt5Gui.so.5 + 0x1342e7)
                        #6  0x00007f80e333e769 _ZN22QGuiApplicationPrivate25createPlatformIntegrationEv (libQt5Gui.so.5 + 0x13e769)
                        #7  0x00007f80e333fa99 _ZN22QGuiApplicationPrivate21createEventDispatcherEv (libQt5Gui.so.5 + 0x13fa99)
                        #8  0x00007f80e2e9b215 _ZN23QCoreApplicationPrivate4initEv (libQt5Core.so.5 + 0x29b215)
                        #9  0x00007f80e333fb46 _ZN22QGuiApplicationPrivate4initEv (libQt5Gui.so.5 + 0x13fb46)
                        #10 0x00007f80e3b75b9e _ZN19QApplicationPrivate4initEv (libQt5Widgets.so.5 + 0x175b9e)
                        #11 0x00007f80e44b7048 _ZN4LXQt11ApplicationC1ERiPPc (liblxqt.so.1 + 0x27048)
                        #12 0x00005607c2b6d07e n/a (lxqt-session + 0xb07e)
                        #13 0x00007f80e2639850 n/a (libc.so.6 + 0x23850)
                        #14 0x00007f80e263990a __libc_start_main (libc.so.6 + 0x2390a)
                        #15 0x00005607c2b6de35 _start (lxqt-session + 0xbe35)

                        Stack trace of thread 584:
                        #0  0x00007f80e2713c0f __poll (libc.so.6 + 0xfdc0f)
                        #1  0x00007f80e2b6cc2f n/a (libglib-2.0.so.0 + 0xb7c2f)
                        #2  0x00007f80e2b0d0e2 g_main_context_iteration (libglib-2.0.so.0 + 0x580e2)
                        #3  0x00007f80e2eea5cc _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2ea5cc)
                        #4  0x00007f80e2e9a484 _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x29a484)
                        #5  0x00007f80e2cf72b6 _ZN7QThread4execEv (libQt5Core.so.5 + 0xf72b6)
                        #6  0x00007f80e41e1a9a n/a (libQt5DBus.so.5 + 0x18a9a)
                        #7  0x00007f80e2cf34da n/a (libQt5Core.so.5 + 0xf34da)
                        #8  0x00007f80e269d44b n/a (libc.so.6 + 0x8744b)
                        #9  0x00007f80e2720e40 n/a (libc.so.6 + 0x10ae40)
                        ELF object binary architecture: AMD x86-64
 sddm-helper[560]: [PAM] Closing session
 sddm-helper[560]: pam_unix(sddm-autologin:session): session closed for user patty
 systemd[1]: systemd-coredump@0-586-0.service: Deactivated successfully.
 sddm-helper[560]: [PAM] Ended.
 sddm[360]: Auth: sddm-helper exited with 11
 sddm[360]: Display server stopping...
...
```
This is on an older laptop, with an Intel Core2 and Radeon graphics.  On this machine, LXQt starts with no problem when using `startx` instead of `sddm`.  But then, other people with sddm 0.20.0-1 are having no problem starting LXQt, so still a little mysterious.

**Comment by Vogtinator** [comment]
This is unrelated. It smells like some other cases where some file like `/etc/profile` overrides `$XAUTHORITY`, which is invalid.

**Comment by kappa-maintainer** [comment]
Same here on latest Debian Sid with *linux-image-amd64/unstable,now 6.4.4-2 amd64*
sddm works fine under 6.3 though
```02 09:44:49 debian systemd-coredump[1565]: [🡕] Process 1562 (sddm) of user 1000 dumped core.
                                                
                                                Module libsystemd.so.0 from deb systemd-254-1.amd64
                                                Stack trace of thread 1562:
                                                #0  0x00007fb1a4ea80fc __pthread_kill_implementation (libc.so.6 + 0x8a0fc)
                                                #1  0x00007fb1a4e5a472 __GI_raise (libc.so.6 + 0x3c472)
                                                #2  0x00007fb1a4e444b2 __GI_abort (libc.so.6 + 0x264b2)
                                                #3  0x00007fb1a549107d _ZNK14QMessageLogger5fatalEPKcz (libQt5Core.so.5 + 0x9107d)
                                                #4  0x000055ab0c04a644 n/a (sddm + 0x11644)
                                                #5  0x000055ab0c08def8 n/a (sddm + 0x54ef8)
                                                #6  0x000055ab0c074968 n/a (sddm + 0x3b968)
                                                #7  0x000055ab0c0863a4 n/a (sddm + 0x4d3a4)
                                                #8  0x000055ab0c0865de n/a (sddm + 0x4d5de)
                                                #9  0x000055ab0c088b3f n/a (sddm + 0x4fb3f)
                                                #10 0x000055ab0c088c83 n/a (sddm + 0x4fc83)
                                                #11 0x00007fb1a56fba32 n/a (libQt5Core.so.5 + 0x2fba32)
                                                #12 0x000055ab0c087d6f n/a (sddm + 0x4ed6f)
                                                #13 0x00007fb1a56fba32 n/a (libQt5Core.so.5 + 0x2fba32)
                                                #14 0x00007fb1a62942bf _ZN23QDBusPendingCallWatcher8finishedEPS_ (libQt5DBus.so.5 + 0x712bf)
                                                #15 0x00007fb1a56ef400 _ZN7QObject5eventEP6QEvent (libQt5Core.so.5 + 0x2ef400)
                                                #16 0x00007fb1a56c30ad _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt5Core.so.5 + 0x2c30ad)
                                                #17 0x00007fb1a56c66d1 _ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData (libQt5Core.so.5 + 0x2c66d1)
                                                #18 0x00007fb1a571d1e3 n/a (libQt5Core.so.5 + 0x31d1e3)
                                                #19 0x00007fb1a4b945e5 g_main_context_dispatch (libglib-2.0.so.0 + 0x595e5)
                                                #20 0x00007fb1a4b948e8 n/a (libglib-2.0.so.0 + 0x598e8)
                                                #21 0x00007fb1a4b9497c g_main_context_iteration (libglib-2.0.so.0 + 0x5997c)
                                                #22 0x00007fb1a571c8d6 _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x31c8d6)
                                                #23 0x00007fb1a56c1b7b _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2c1b7b)
                                                #24 0x00007fb1a56ca020 _ZN16QCoreApplication4execEv (libQt5Core.so.5 + 0x2ca020)
                                                #25 0x000055ab0c04ddf1 n/a (sddm + 0x14df1)
                                                #26 0x00007fb1a4e456ca __libc_start_call_main (libc.so.6 + 0x276ca)
                                                #27 0x00007fb1a4e45785 __libc_start_main_impl (libc.so.6 + 0x27785)
                                                #28 0x000055ab0c04e0b1 n/a (sddm + 0x150b1)
                                                
                                                Stack trace of thread 1563:
                                                #0  0x00007fb1a4f199ef __GI___poll (libc.so.6 + 0xfb9ef)
                                                #1  0x00007fb1a4b9485e n/a (libglib-2.0.so.0 + 0x5985e)
                                                #2  0x00007fb1a4b9497c g_main_context_iteration (libglib-2.0.so.0 + 0x5997c)
                                                #3  0x00007fb1a571c8d6 _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x31c8d6)
                                                #4  0x00007fb1a56c1b7b _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2c1b7b)
                                                #5  0x00007fb1a54d8dde _ZN7QThread4execEv (libQt5Core.so.5 + 0xd8dde)
                                                #6  0x00007fb1a623a757 n/a (libQt5DBus.so.5 + 0x17757)
                                                #7  0x00007fb1a54d9ffd n/a (libQt5Core.so.5 + 0xd9ffd)
                                                #8  0x00007fb1a4ea63ec start_thread (libc.so.6 + 0x883ec)
                                                #9  0x00007fb1a4f26a1c __clone3 (libc.so.6 + 0x108a1c)
                                                ELF object binary architecture: AMD x86-64```

**Comment by noraj** [comment]
PS : we are currently bisecting on ArchLinux https://bugs.archlinux.org/task/79017?project=1&string=sddm

**Comment by Vogtinator** [comment]
FTR, I've been running 6.4 for quite a while now, even in the RC phase. On a different Laptop I'm at 6.4.6 and I haven't had any issues.

> Same here on latest Debian Sid with linux-image-amd64/unstable,now 6.4.4-2 amd64

That shows that it logged the fatal error to the journal. Do you have that message?

**Comment by davidedmundson** [comment]
>_ZNK14QMessageLogger5fatalEPKcz

We only have one qFatal in SDDM code:
common/XAuth.cpp:        qFatal("Failed to create xauth file");

Which implies `RUNTIME_DIR`  which comes from cmake points to a directory we don't have permissions to make

**Comment by noraj** [comment]
> FTR, I've been running 6.4 for quite a while now, even in the RC phase. On a different Laptop I'm at 6.4.6 and I haven't had any issues.
> 
> > Same here on latest Debian Sid with linux-image-amd64/unstable,now 6.4.4-2 amd64
> 
> That shows that it logged the fatal error to the journal. Do you have that message?

For what I have digged and understood seems to be related to some AMD graphic drivers of some model of GPU only, so it's totally normal there is no issue for 99% of people.

**Comment by kappa-maintainer** [comment]
> > FTR, I've been running 6.4 for quite a while now, even in the RC phase. On a different Laptop I'm at 6.4.6 and I haven't had any issues.
> > > Same here on latest Debian Sid with linux-image-amd64/unstable,now 6.4.4-2 amd64
> > 
> > 
> > That shows that it logged the fatal error to the journal. Do you have that message?
> 
> For what I have digged and understood seems to be related to some AMD graphic drivers of some model of GPU only, so it's totally normal there is no issue for 99% of people.

I am using RX6650XT + r53600X with mesa driver 23.1.4, how can I help?

**Comment by tbergman99** [comment]
I get this issues with 7900xt+12700k. Just adding this as it seems to be hardware related

**Comment by kappa-maintainer** [comment]
Further tests on greetd gave me an 0 GPU error, while lxdm can't recognize my monitor
I think I can say some changes from 6.4 break the graphic driver

**Comment by noraj** [comment]
Git bisect results on the kernel available here: https://bugs.archlinux.org/task/79017#comment221080
